goldmane/pkg/aggregator/aggregator.go

```diff
@@ -317,6 +322,10 @@ func (a *LogAggregator) rollover() time.Duration {
 		return nil
 	})
 
+	// Tell the stream manager that we've rolled over. This will trigger emission of any
+	// flows that are ready to be emitted to active stream clients.
+	a.streams.rollover(a.buckets, a.diachronics)
```

Comment:
Right now this is synchronous with the rest of the rollover processing - TBD if this is good enough. It has the nice effect of not needing to interact with core LogAggregator memory from another goroutine, but it does means we'll service every stream sequentially from the main loop.

We could also take a snapshot of the memory and send it via a channel, and run the stream manager on a second goroutine, which I believe is closer to how your original commit worked @Brian-McM 

Main reason I didn't just start with that approach is that the streams are tied to the rollover interval (we're emitting aggregated flows), whereas in your original PR we were siphoning off pre-aggregation Flows.

Reply:
Ultimately I don't want to over optimize this implementation at this stage - it's still not even clear what the final requirements for API and behavior will be. i.e., will we need the ability to start a stream at a particular point in time and catch-up from there? Will we need configurable aggregation windows? etc.